### PR TITLE
refactor (jkube-kit) : Move logic for adding OpenShift Build specific properties to DefaultGeneratorManager

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

In Resource mojo and tasks, we've duplicated code that adds the following properties to JavaProject in case of OpenShift mode:
- `jkube.image.user` : Sets image user property to currently logged in OpenShift namespace so that image user portion points to namespace
- `jkube.internal.effective.platform.mode` : Used by some enrichers to determine whether current platform is OpenShift. This looks like a remnant of Fabric8 Maven Plugin refactor. We should get rid of this in follow up PR.

This PR only moves code from Mojos/Tasks -> DefaultGeneratorManager.generateAndMerge()

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] My code follows the style guidelines of this project
 - [X] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [ ] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [X] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [X] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [X] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3037 (_2024-05-09T13:41:58Z_)

:gear: [JKube E2E Tests (9018201936)](https://github.com/eclipse-jkube/ci/actions/runs/9018201936) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3037)](https://github.com/eclipse-jkube/ci/actions/runs/9018201936/job/24778270094)

**Comment by manusa** [comment]
I'll rebase this once https://github.com/eclipse-jkube/jkube/pull/3042 is in.
I'd like to change a few things.

File: gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 import java.util.List;
 import java.util.Optional;
 
+import com.networknt.schema.utils.StringUtils;
```
Comment (rohanKanojia) [inline gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java:24]: You've imported wrong StringUtils

Comment (manusa) [reply gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java:24]: :facepalm:

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**